snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,1035 @@
+  - jhtarballs: the directory on the host containing source tarballs
+  - jhmirror: the directory on the host containing DVCS mirror repositories
+  - jhcheckout: the directory on the host container DVCS checkout repositories
+  - ccache: the directory on the host to use for ccache caching
+  - disable-pull: True to skip the pull stage (default: False)
+  - disable-build: True to skip the build stage (default: False)
+  - disable-jhupdate: True to skip updating jhtarballs, jhmirror and jhcheckout
+                      (default: False)
+  - disable-parallel: True to disable parallel building (default: False)
+
+Advice:
+
+  - The command for launching an app from a JHBuild part is:
+    jhbuild -f $SNAP/etc/jhbuildrc run <app>
+
+    So for a glade snap, the app would look like:
+
+    apps:
+      glade:
+        command: jhbuild -f $SNAP/etc/jhbuildrc run glade
+        plugs: [x11]
+
+  - Building WebKit requires a lot of time and computing power. If WebKit is a
+    dependency of your JHBuild module, it can be skipped by adding '-WebKit' to
+    the list of modules, and adding 'libwebkit2gtk-4.0-dev' to the list of
+    container-packages.
+
+  - Using the debmirror option in combination with a local apt-cacher-ng
+    installation can prevent repeated downloading of .deb packages. Use the IP
+    address of your lxcbr0 interface with the port number apt-cacher-ng uses.
+    For example, debmirror: http://10.0.3.1:3142
+
+  - Specify directories on your host for jhtarballs and jhmirror to prevent
+    repeated downloading of the JHBuild module sources. It is best to reserve
+    directories on your local machine that can be used for all snaps you might
+    want to build.
+
+  - If you are debugging your snap, but don't need to rebuild the sources (for
+    example, to determine the list of plugs required under confinement), use
+    the disable-pull, disable-build, and disable-jhupdate options.
+
+  - To add a shell for debugging your snap, add an extra app whose command is:
+    jhbuild -f $SNAP/etc/jhbuildrc shell. You might also want to add gdb,
+    strace, etc. to your list of stage-packages for debugging purposes.
+"""
+
+import logging
+import os
+import shlex
+import platform
+import snapcraft
+import time # TODO: wait for networking
+
+logger = logging.getLogger(__name__)
+
+BUILD_PACKAGES = [
+    'ca-certificates',
+    'git',
+    'make',
+    'autoconf',
+    'automake',
+    'gettext',
+    'pkg-config',
+    'yelp-tools',
+    'ccache',
+    'libtool',
+    'docbook-xsl',
+    'libxml-parser-perl',
+    'cvs',
+    'subversion',
+    'flex',
+    'bison',
+    'apt-file',
+]
+
+STAGE_PACKAGES = [
+    'python',
+    'ttf-ubuntu-font-family',
+    'adwaita-icon-theme-full',
+]
+
+# build dependencies that JHBuild doesn't know about
+MODULE_EXCEPTIONS = {
+    'vala':                                                     ['valac'],
+    'gst-plugins-bad':                                          ['libgl1-mesa-dev'],
```

Comment:
I don't think it's common in python to use spaces like this, to format a table. I would prefer just one space between the : and the element. You might even get pep8 errors for this, I'm not sure.